_posts/2021-01-09-Julia-1.6-what-has-changed-since-1.0.md

```diff
@@ -0,0 +1,639 @@
+
+
+### Code highlighting for `code_llvm` and `code_native`
+This was added in [#36634](https://github.com/JuliaLang/julia/issues/36634).
+This functionality was first implemented in the [ColoredLLCodes.jl](https://github.com/kimikage/ColoredLLCodes.jl), where it worked by monkey-patching the InteractiveUtils stdlib.
+That package does still work on julia 1.0.
+
+
+**Julia 1.0:**
+
+![Julia 1.0 code_llvm]({{site.url}}/posts_assets/Julia-1.0-1.6-changes/julia-1.0-code-llvm.png)
+
+**Julia 1.6:**
+
+![Julia 1.6 code_llvm]({{site.url}}/posts_assets/Julia-1.0-1.6-changes/julia-1.6-code-llvm.png)
+
+The REPL itself still doesn't have syntax highlighting for Julia code though.
+The [OhMyRepl.jl](https://github.com/KristofferC/OhMyREPL.jl) package does provide that, and works all versions of Julia.
+It is a lot more than a series of regex though, so I don't think we are going to see it built into julia too soon.
+Probably one day, though, as its big dependencies are also required if the parser wants to move to be written in julia.
+(Though I also don't expect that any time soon)
+
+### Clearer Stacktraces
+
+Just like colored `code_llvm` colored stack-traces also originated in a package that was doing some nasty monkey-patching: [ClearStacktrace.jl](https://github.com/jkrumbiegel/ClearStacktrace.jl).
+This was added into Base itself in [#36134](https://github.com/JuliaLang/julia/pull/36134)
+
+**Julia 1.0:**
+
+![Julia 1.0 stacktrace]({{site.url}}/posts_assets/Julia-1.0-1.6-changes/julia-1.0-stacktrace.png)
+
+**Julia 1.6:**
+
+![Julia 1.6 stacktrace]({{site.url}}/posts_assets/Julia-1.0-1.6-changes/julia-1.6-stacktrace.png)
+
+The first thing you probably notice is the colored package names, to make it more clear where the error is coming from.
+You also should notice the dimming of type-parameters to make complicated types easier to read.
+Also the addition of argument names, and showing keyword arguments bearing functions as they are written, rather than as a weird `#DataFrame#654` internal function.
+The whole thing just looks more modern and polished.
+
+### Time Traveling Debugger
+
+[Julia 1.5 added built in support](https://julialang.org/blog/2020/05/rr/) for [`rr`](https://rr-project.org/) (linux only for now).
+This is not useful so much for debugging julia code, but for debugging julia itself.
+If you get into one of the very rare situations where the language is doing something truly nonsense; you can send a bug report recorded via `rr`, and someone can debug *exactly* what is happening.
+Event to the extent of [diagnosing faulty RAM](https://julialang.org/blog/2020/09/rr-memory-magic/).
+`rr` is an impressively cool piece of tech; it boils down to the fact that linux's `fork` is incredibly cheap, because it take advantage of [copy-on-write](https://en.wikipedia.org/wiki/Copy-on-write); so it basically `fork`s before every system call, and serializes what is going on; so you have a full track of everything that happened.
```

Comment:
```suggestion
`rr` is an impressively cool piece of tech; it boils down to the fact that linux's `fork` is incredibly cheap, because it takes advantage of [copy-on-write](https://en.wikipedia.org/wiki/Copy-on-write); so it basically `fork`s before every system call, and serializes what is going on; so you have a full track of everything that happened.
```